components/database.ts

```diff
@@ -6,22 +6,12 @@ async function dbConnect() {
   // connecting or disconnecting (readyState 1, 2 and 3)
   if (mongoose.connection.readyState >= 1) return
   console.log(`Connecting to ${process.env.NODE_ENV} database...`)
-  return mongoose
-    .connect(process.env.MONGO_URL, {
-      // useNewUrlParser: true,
-      // useUnifiedTopology: true,
-      // useFindAndModify: false,
-      // useCreateIndex: true,
-    })
-    .then(() => {
-      console.log('Connected to database.')
-    })
-    .catch(err => {
-      console.error(`Couldn't connect to database`, err)
-      return undefined
-    })
+  try{
+    await  mongoose.connect(process.env.MONGO_URL);
+    console.log("Connected to database");
+  }catch(err){
+    throw `Error connecting to database: ${err}`; 
+  }
 }
 
-let connectionPromise = dbConnect()
-
-export default connectionPromise
+export default dbConnect;
```

Comment:
This definitely needs to be reverted. Now the `dbConnect` function is returned, instead of a promise. So when in other components we call `await dbConnect()`, it'll re-establish mongo connection every time, instead of returning the existing mongo connection. Very inefficient and results in a memory leak + connection pool will me maxed out very quickly. I've made this mistake before.